src/schematools/types.py

```diff
@@ -527,14 +498,31 @@ def use_dimension_fields(self) -> bool:
         return self._parent_schema.use_dimension_fields
 
     @property
-    def temporal(self) -> Optional[Dict[str, Union[str, Dict[str, List[str]]]]]:
-        """Return the temporal info from the dataset schema """
-        return self._parent_schema.fetch_temporal(field_modifier=lambda x: x)
+    def temporal(
+        self, field_modifier: Optional[Callable] = None
```

Comment:
It would help if the `Callable` was more specific. What does the `Callable` accept as arguments and what does it return?

Reply:
hoe kan een property ook nog een parameter ontvangen?

Reply:
Hier liep ik ook net tegenaan toen ik het lokaal testte. Vreemd eigenlijk dat python dit toestaat.

Reply:
Damn, I missed this was defined as a property. Yes, a property can't have a (usable) parameter.

Reply:
Well I know someone else who didn't notice. :-) Still weird Python allows it. Can anyone think of a good reason for this? I removed the argument now. It doesnt seem to be needed anymore.

Reply:
Maybe someone had the following use case in mind:

```python
dataset: DatasetSchema = ...
...
temporal_meuk = DatasetSchema.fetch_temporal.fget(dataset, field_modifier=...)
```

Reply:
That's exactly the usecase. The `field_modifier` was added to be backwards compatible. But, in hindsight, it would be better to remove it, it only complicates things and makes the code less readable.
The original implementation was doing an implicit `to_snake_case` on the result. Actually, it would be better to return the unaltered temporal info and leave it to the caller to do casing operations if needed.